### PR TITLE
Added new BuildCause, for my nuget trigger

### PR DESCRIPTION
**Pull request by baritonehands**
Added new BuildCause, for my nuget trigger

### CODE REVIEW

File: src/main/java/org/jenkins_ci/plugins/run_condition/core/CauseCondition.java [hunk 1]
```diff
@@ -66,15 +68,17 @@ public boolean isCausedBy(String className) {
 
         // if XTrigger plugin is installed:
         // file change
-        FS_CAUSE("org.jenkinsci.plugins.fstrigger.FSTriggerCause", "FSTrigger"),
+        FS_CAUSE(XTriggerCause.class, "FSTrigger"),
```
Comment (imod) [inline src/main/java/org/jenkins_ci/plugins/run_condition/core/CauseCondition.java]: why did you change this to use the class instead of the class name as a String? now the plugin has a dependency to the `xtrigger` plugin and everyone using this plugin needs to install that plugin too. Please change this back to use the full qualified name of the class as a String

Comment (baritonehands) [inline src/main/java/org/jenkins_ci/plugins/run_condition/core/CauseCondition.java]: It's been awhile, but I believe it was because the XTrigger integration
didn't actually work with the strings. They were always an XTriggerCause
class no matter what triggered it.

On Wed, Mar 30, 2016 at 2:38 AM Dominik Bartholdi <notifications@github.com>
wrote:

> In
> src/main/java/org/jenkins_ci/plugins/run_condition/core/CauseCondition.java
> <https://github.com/jenkinsci/run-condition-plugin/pull/9#discussion_r57842590>
> :
>
> > @@ -66,15 +68,17 @@ public boolean isCausedBy(String className) {
> >
> >          // if XTrigger plugin is installed:
> >          // file change
> > -        FS_CAUSE("org.jenkinsci.plugins.fstrigger.FSTriggerCause", "FSTrigger"),
> > +        FS_CAUSE(XTriggerCause.class, "FSTrigger"),
>
> why did you change this to use the class instead of the class name as a
> String? now the plugin has a dependency to the xtrigger plugin and
> everyone using this plugin needs to install that plugin too. Please change
> this back to use the full qualified name of the class as a String
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/run-condition-plugin/pull/9/files/929a10d1248b9224b895ff6f7eccfa6d9ed7dc68#r57842590>
>
-- 
Typed on a tiny keyboard.

Comment (Greybird) [inline src/main/java/org/jenkins_ci/plugins/run_condition/core/CauseCondition.java]: Hi,

In fact, @baritonehands you are right.
If you look at the build xml, you get this:
```
<?xml version='1.0' encoding='UTF-8'?>
<build>
  <actions>
    <org.jenkinsci.lib.xtrigger.XTriggerCauseAction plugin="fstrigger@0.39">
      <build class="build" reference="../../.."/>
    </org.jenkinsci.lib.xtrigger.XTriggerCauseAction>
    <hudson.model.CauseAction>
      <causes>
        <org.jenkinsci.lib.xtrigger.XTriggerCause plugin="fstrigger@0.39">
          <triggerName>FSTrigger</triggerName>
          <causeFrom>Triggered by a change to a file</causeFrom>
          <logEnabled>true</logEnabled>
        </org.jenkinsci.lib.xtrigger.XTriggerCause>
      </causes>
    </hudson.model.CauseAction>
  </actions>
</build>

```
So the plugin is actually not working with the plugins linked to XTrigger.

Also note that the **isCausedBy** method has been updated to take the trigger name into account, which is the second part of the correction for xtrigger causes.
As this creates the needs the xtrigger **lib** (not the xtrigger plugin), this makes no sense to stay with a string here, IMHO. Of course, reflection is still an option.

Regards,

Arnaud

**Comment by timja** [comment]
Merged in https://github.com/jenkinsci/run-condition-plugin/pull/14
